### PR TITLE
#2949 Attachment points should be within S-Group brackets

### PR DESCRIPTION
**Pull request by StarlaStarla**
## How the feature works? / How did you fix the issue?
(Screenshots, videos, or GIFs, if applicable)

## Check list
- [x] unit-tests written
- [x] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"

### CODE REVIEW

File: packages/ketcher-core/src/domain/entities/sgroup.ts [hunk 1]
```diff
@@ -504,8 +505,18 @@ export class SGroup {
       });
       braketBox = !braketBox ? bbb : Box2Abs.union(braketBox, bbb!);
     });
-    const vext = new Vec2(0.2, 0.4);
-    if (braketBox) braketBox = (braketBox as Box2Abs).extend(vext, vext);
+    if (!render) render = global._ui_editor.render;
```
Comment (Nitvex) [inline packages/ketcher-core/src/domain/entities/sgroup.ts:508]: Could you, please, describe the case when render is not available?

**Review by krakmielepam — APPROVED** [review]
Besides other comments I approve

File: packages/ketcher-core/src/domain/entities/sgroup.ts (hunk 1 above)
Comment (StarlaStarla) [reply packages/ketcher-core/src/domain/entities/sgroup.ts:508]: when copy or cut a s-group in on the clipboard

File: ketcher-autotests/tests/R-group-tool/r-group-fragment-tool.spec.ts [hunk 2]
```diff
@@ -121,6 +126,44 @@ test.describe('Open Ketcher', () => {
     await clickModalButton(page, 'Apply');
   });
 
+  test('Brackets rendering for atom', async ({ page }) => {
+    await openFileAndAddToCanvas('simple-chain.ket', page);
+    await selectNestedTool(page, RgroupTool.R_GROUP_FRAGMENT);
+    await clickOnAtom(page, 'C', atomNumber);
+    await page.getByText('R5').click();
```
[the remark this answers is not in the document]
Comment (StarlaStarla) [reply ketcher-autotests/tests/R-group-tool/r-group-fragment-tool.spec.ts]: done for this

File: ketcher-autotests/tests/utils/canvas/tools/selectNestedTool/selectNestedTool.ts [hunk 3]
```diff
@@ -9,22 +9,25 @@ import { getToolType, openTool } from './helpers';
  *
  * @param page - playwright page object
  * @param toolElementId - tuple [presses: number, domElementId: string]
+ * @param currentToolId - current selectede-group tool
```
Comment (Nitvex) [inline ketcher-autotests/tests/utils/canvas/tools/selectNestedTool/selectNestedTool.ts:12]: Here is a small typo

File: ketcher-autotests/tests/utils/canvas/tools/selectNestedTool/selectNestedTool.ts [hunk 4]
```diff
@@ -9,22 +9,25 @@ import { getToolType, openTool } from './helpers';
...
  * presses - specifies the number of times to press Tab key
  * domElementId - id of the element to search
  */
 export const selectNestedTool = async (
   page: Page,
   toolElementId: [presses: number, domElementId: string],
+  currentToolId?: [presses: number, domElementId: string],
```
Comment (Nitvex) [inline ketcher-autotests/tests/utils/canvas/tools/selectNestedTool/selectNestedTool.ts:19]: And could you, please, clarify what it is used for?

Comment (StarlaStarla) [reply ketcher-autotests/tests/utils/canvas/tools/selectNestedTool/selectNestedTool.ts:19]: When we create a r-group with fragments, we need to switch the r-group tool from ATTACHMENT_POINTS to R_GROUP_FRAGMENT. If we didnt pass in this ATTACHMENT_POINTS as a param, selectNestedTool would call openTool func to query the default element with testId of 'rgroup-label', while actually there is no such element at this point, only 'rgroup-attpoints' exist.
image

**Review by rrodionov91 — APPROVED** [review]

**Pull request merged**